### ISSUE TITLE
Bug: dropout is not disabled during validation

### ISSUE THREAD

**Issue by dbbert** [vlfeat/matconvnet#327]
Lines 13 and 25 in matlab/+dagnn/DropOut.m should be changed from
    if strcmp(obj.net.mode, 'test')
to
    if strcmp(obj.net.mode, 'test') || strcmp(obj.net.mode, 'val')
so that dropout also gets disabled during the validation run.

**Comment by vedaldi** [comment]
Hi,

there are only two modes defined, either ‘normal’, or ‘test’. The latter mode disables or bypasses certain layers (dropout and in the next beta17 batch normalisation).

There is no reason to distinguish between `test’ and `validation’ at this level (i.e. in validation one can choose to use either `normal’ or `test’ mode, depending on the desired effect — both make sense).

Best wishes,

Andrea

**Comment by nightrome** [comment]
I agree that this is a good concept, but then the cnn_train_dag example should also change the net.mode to "test" for validation.

**Comment by vedaldi** [comment]
Hi, yes, that’s right and was an oversight on our part. beta17 will do this. We should be releasing it in the next day or two.
